### PR TITLE
yolov5 base class renamed

### PR DESCRIPTION
**Pull request by shaydeci**
First PR as a part of integrating Yolox into SG, holds:

Yolo base class for the other yolo models to inherit from.
Also shoved the pycocotools requirement into this PR.

### CODE REVIEW

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 1]
```diff
@@ -27,19 +27,29 @@ def width_multiplier(original, factor, divisor: int = None):
...
+            width_mult = lambda channels: width_multiplier(channels, width_mult_factor, 8)
+        elif yolo_version == 'v3.0':
+            struct = (3, 9, 9, 3)
+            block = BottleneckCSP
+            activation_type = nn.Hardswish
+            width_mult = lambda channels: width_multiplier(channels, width_mult_factor)
+        else:
+            raise NotImplementedError(f'YoloV5 release version {yolo_version} is not supported, use one of: '
+                                      f'"v3.0", "v6.0"')
+    elif yolo_type == 'yoloX':
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/models/detection_models/csp_darknet53.py:45]: I really think it'd be a bad idea to take it out from here...

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 2]
```diff
@@ -211,14 +222,14 @@ def __init__(self, arch_params: HpmStruct):
         self._modules_list.append(Conv(width_mult(256), width_mult(512), 3, 2, activation_type))                     # 5
         self._modules_list.append(block(width_mult(512), width_mult(512), struct[2], activation_type))               # 6
         self._modules_list.append(Conv(width_mult(512), width_mult(1024), 3, 2, activation_type))                    # 7
-        if yolo_version == 'v6.0':
-            self._modules_list.append(block(width_mult(1024), width_mult(1024), struct[3], activation_type))         # 8
-            self._modules_list.append(SPPF(width_mult(1024), width_mult(1024), 5, activation_type))                  # 9
-        elif yolo_version == 'v3.0':
+        if yolo_type == 'yoloX' or yolo_version == 'v3.0':
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/models/detection_models/csp_darknet53.py:225]: I really think it'd be a bad idea to take it out from here...

**Review by ofrimasad — APPROVED** [review]
LGTM
after discussion, the mentioning of yolox will remain in this PR, but it will be merged only after the other YoloX PRs are ready

**Comment by shaydeci** [comment]
> LGTM
> after discussion, the mentioning of yolox will remain in this PR, but it will be merged only after the other YoloX PRs are ready

We actually need to merge this one first, before the Yolox code PR (as stated in https://github.com/Deci-AI/super-gradients/pull/210#issue-1255355502)

**Pull request merged**